tests/tracing.rs

```diff
@@ -0,0 +1,146 @@
+    /// Create a new `MockWriter` that writes into the specified buffer (behind a mutex).
+    pub fn new(buf: Arc<Mutex<Vec<u8>>>) -> Self {
+        Self { buf }
+    }
+
+    /// Give access to the internal buffer (behind a `MutexGuard`).
+    fn buf(&self) -> io::Result<MutexGuard<Vec<u8>>> {
+        // Note: The `lock` will block. This would be a problem in production code,
+        // but is fine in tests.
+        self.buf
+            .lock()
+            .map_err(|_| io::Error::from(io::ErrorKind::Other))
+    }
+}
+
+impl io::Write for MockWriter {
+    fn write(&mut self, buf: &[u8]) -> io::Result<usize> {
+        // Lock target buffer
+        let mut target = self.buf()?;
+
+        // Write to stdout in order to show up in tests
+        print!("{}", String::from_utf8(buf.to_vec()).unwrap());
+
+        // Write to buffer
+        target.write(buf)
+    }
+
+    fn flush(&mut self) -> io::Result<()> {
+        self.buf()?.flush()
+    }
+}
+
+impl MakeWriter<'_> for MockWriter {
+    type Writer = Self;
+
+    fn make_writer(&self) -> Self::Writer {
+        MockWriter::new(self.buf.clone())
+    }
+}
+
+/// Return a new subscriber that writes to the specified [`MockWriter`].
+///
+/// [`MockWriter`]: struct.MockWriter.html
+pub fn get_subscriber(mock_writer: MockWriter, env_filter: &str) -> Dispatch {
+    FmtSubscriber::builder()
+        .with_env_filter(env_filter)
+        .with_writer(mock_writer)
+        .with_level(true)
+        .with_ansi(false)
+        .without_time()
+        .into()
+}
+
+pub fn with_logs<F>(buf: &Mutex<Vec<u8>>, f: F)
+where
+    F: Fn(&[&str]),
+{
+    let buf = buf.lock().unwrap();
+    let logs: Vec<&str> = std::str::from_utf8(&buf)
+        .expect("Logs contain invalid UTF8")
+        .lines()
+        .collect();
+    f(&logs)
+}
+
+struct Tracer;
+
+#[async_trait]
+impl Actor for Tracer {
+    type Stop = ();
+    async fn stopped(self) {}
+}
+
+struct Hello(&'static str);
+
+#[async_trait]
+impl Handler<Hello> for Tracer {
+    type Return = ();
+
+    async fn handle(&mut self, message: Hello, _ctx: &mut Context<Self>) {
+        tracing::info!("Hi {}", message.0)
+    }
+}
+
+#[tokio::test]
+async fn assert_send_is_child_of_span() {
+    let buf = Arc::new(Mutex::new(vec![]));
+    let mock_writer = MockWriter::new(buf.clone());
+    let subscriber = get_subscriber(mock_writer, "tracing=trace,xtra=trace");
+    let _g = tracing::dispatcher::set_default(&subscriber);
+
+    let (addr, mut ctx) = Context::<Tracer>::new(None);
+    let _ = addr
+        .send(Hello("hi"))
+        .split_receiver()
+        .instrument(tracing::info_span!("Test span"))
+        .await;
+    ctx.yield_once(&mut Tracer).await;
+
+    with_logs(&buf, |lines: &[&str]| {
+        assert_eq!(
+            lines,
+            [
+                r#" INFO Test span:xtra_actor_request{actor="tracing::Tracer" tracing::Hello}:xtra_message_handler{actor="tracing::Tracer" tracing::Hello}: tracing: Hi hi"#
```

Comment:
I think we can actually see here that using `message` as a span variable name has weird side effects. Should maybe call it `msg_type` or something?

Reply:
That's a good point, actually. It seems to work okay with Jaeger (and Grafana Tempo), so I never noticed this

Reply:
Speaking of this, I was thinking of removing the tags from everything but `xtra_actor_request`, otherwise they repeat. Should this and the rename go in a separate PR?

Reply:
> Speaking of this, I was thinking of removing the tags from everything but `xtra_actor_request`, otherwise they repeat. Should this and the rename go in a separate PR?

I'd prefer a 2nd PR yeah :)